linera-core/src/client.rs

```diff
@@ -1328,6 +1335,59 @@ where
                         message.action = MessageAction::Reject;
                         continue;
                     }
+                } else if let ChainError::ExecutionError(
+                    ExecutionError::SystemError(SystemExecutionError::BlobNotFoundOnRead(blob_id)),
+                    _,
+                ) = &**chain_error
+                {
+                    // Even after synching this chain with the validators, we still don't have the blob we're trying to read.
+                    // Which means it was published by a different chain, so we need to update this chain with it.
+                    let validators = self.validator_nodes().await?;
+                    let mut last_used_by = HashMap::new();
+                    for (_, mut node) in validators.into_iter() {
+                        if let Ok(last_used_hash) = node.blob_last_used_by(*blob_id).await {
+                            match last_used_by.entry(last_used_hash) {
+                                hash_map::Entry::Vacant(v) => {
+                                    v.insert(vec![node]);
+                                }
+                                hash_map::Entry::Occupied(mut o) => {
+                                    o.get_mut().push(node);
+                                }
+                            }
+                        }
+                    }
+
+                    ensure!(
+                        !last_used_by.is_empty(),
+                        ChainClientError::BlobNotFound(*blob_id)
+                    );
+                    ensure!(
+                        last_used_by.len() == 1,
+                        ChainClientError::ConflictingLastUsageData(*blob_id)
```

Comment:
Why would the validators agree on that?
It suffices for _some_ validator to provide us with any certificate that uses the blob.

Reply:
True, they wouldn't. I guess this failing would indicate larger protocol issues that will probably be caught by other errors 😅 will remove

Reply:
(To clarify: In normal operation, especially for busy blobs, validators will often have different `last_used_by` entries for the same blob. That is expected.)